### PR TITLE
feat: enable midpoint inside linear element editor

### PR DESCRIPTION
**Pull request by ad1992**
https://user-images.githubusercontent.com/11256141/184318654-ad088f7b-f7bc-4f25-8fb0-41739fd55783.mp4

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -3670,6 +3670,7 @@ class App extends React.Component<AppProps, AppState> {
           if (hitLinkElement) {
             return false;
           }
+
```
Comment (ad1992) [inline src/components/App.tsx:3673]:
```suggestion
```

File: src/renderer/renderScene.ts [hunk 2]
```diff
@@ -205,32 +203,60 @@ const renderLinearPointHandles = (
...
   }
+  const { POINT_HANDLE_SIZE } = LinearElementEditor;
+
   points.forEach((point, idx) => {
     const isSelected =
       !!appState.editingLinearElement?.selectedPointsIndices?.includes(idx);
-    renderSingleLinearPoint(context, appState, renderConfig, point, isSelected);
+    const radius = appState.editingLinearElement
+      ? POINT_HANDLE_SIZE
+      : POINT_HANDLE_SIZE / 2;
```
Comment (dwelle) [inline src/renderer/renderScene.ts:213]: Can be kept outside — why calculate this inside the loop, and on two separate places?

File: src/renderer/renderScene.ts [hunk 3]
```diff
@@ -205,32 +203,60 @@ const renderLinearPointHandles = (
...
+        highlightPoint(centerPoint, context, renderConfig);
+        renderSingleLinearPoint(
+          context,
+          appState,
+          renderConfig,
+          centerPoint,
+          radius,
+          false,
+        );
+      }
```
Comment (dwelle) [inline src/renderer/renderScene.ts:252]: what's the difference between the branches? If either of those callbacks have side effects that work different based on order, then that's not great :D

File: src/renderer/renderScene.ts (hunk 2 above)
Comment (ad1992) [reply src/renderer/renderScene.ts:213]: ofcourse! I dunno why I was doing that 🤦‍♀️

File: src/renderer/renderScene.ts (hunk 3 above)
Comment (ad1992) [reply src/renderer/renderScene.ts:250]: Its not about side effects but the final outcome
for editor -> the point radius is small but hover state is bigger and is on top of the point (as its full violet) so original point is not visible hence I am first rendering point and then hover state

Screenshot 2022-08-18 at 7 31 26 PM

outside editor -> The hover state has bigger radius but the original point is also visible as well hence  drawing the point after the hover state so its visible

Screenshot 2022-08-18 at 7 31 02 PM

Comment (dwelle) [reply src/renderer/renderScene.ts:250]: Mhm... I'd probably make the highlight part of `renderSingleLinearPoint()` instead. But barring that, let's add doc for this so it's clear.

**Review by dwelle — APPROVED** [review]

File: src/renderer/renderScene.ts (hunk 3 above)
Comment (ad1992) [reply src/renderer/renderScene.ts:250]: `highlight` I didn't make part of `renderSelectionElement` as highlight is only for active point and different for outside vs inside editor so kept the logic separate. Will add the doc.

**Pull request merged**